src/ReportManager/DetailsSection/SchemaForm/fields/Text/index.js

```diff
@@ -0,0 +1,81 @@
+import React, { useContext } from 'react';
+
+import useFieldDetails from '../../SchemaFormContext/useFieldDetails';
+import { SchemaFormContext } from '../../SchemaFormContext';
+import { isFieldRequired } from '../../utils';
+
+import styles from './styles.module.scss';
+
+const ShortTextInput = ({ textFieldDetails, fieldName, onChange, className }) => (
+  <input id={fieldName}
+           type="text"
+           value={textFieldDetails.value || textFieldDetails.defaultInput}
+           placeholder={textFieldDetails.placeholder}
+           onChange={onChange}
+           data-testid={`schema-form-short-text-field-input-${fieldName}`}
+           className={className} />
+);
+
+const LongTextInput = ({ textFieldDetails, fieldName, onChange, className }) => (
+  <textarea id={fieldName}
+            value={textFieldDetails.value || textFieldDetails.defaultInput}
+            placeholder={textFieldDetails.placeholder}
+            onChange={onChange}
+            data-testid={`schema-form-long-text-field-input-${fieldName}`}
+            className={className}>
+  </textarea>
+);
+
+const INPUT_TYPE = {
+  SHORT: 'SHORT_TEXT',
+  LONG: 'LONG_TEXT'
+};
+
+const TEXT_INPUT_TYPE_TO_INPUT = {
+  [INPUT_TYPE.SHORT]: ShortTextInput,
+  [INPUT_TYPE.LONG]: LongTextInput,
+};
+
+const VALIDATION_ERROR_TYPES = {
+  REQUIRED: 'required'
+};
+
+export const TextFieldValidators = {
+  [VALIDATION_ERROR_TYPES.REQUIRED]: (fieldValue, fieldName, schema) => {
+    return isFieldRequired(fieldName, schema) ? !fieldValue : false;
+  }
+};
+
+const Text = ({ fieldName }) => {
+  const textFieldDetails = useFieldDetails(fieldName);
+  const label = textFieldDetails.isRequired ? `${textFieldDetails.label} *` : textFieldDetails.label;
+  const TextInput = TEXT_INPUT_TYPE_TO_INPUT[textFieldDetails.inputType];
+  const { onFieldChange } = useContext(SchemaFormContext);
+
+  const stylesTheme = {
```

Comment:
This variable doesn't have any dependency on the content of the component. We should declare it as a constant outside of the component so it's not attached to its lifecycle and its not redefined in every render 👍